onnx/defs/tensor_util.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+#include "tensor_util.h"
+#include <vector>
+#include "onnx/common/platform_helpers.h"
+
+namespace ONNX_NAMESPACE {
+
+#define DEFINE_PARSE_DATA(type, typed_data_fetch)                          \
```

Comment:
Why do you need this function? Is it for processing int64 `pads`? Where is the old code that runs before?

Reply:
There is no such code needed to run before. Previously, `pads` was an attribute and hence there were semantics already in place to parse the data within the attribute. Now `pads` is an input tensor and we need this code to parse the data within the struct now. Hence, this code is needed and used within the optimizer code.

Reply:
Can they be merged into one function?

Reply:
I feel this is maybe not a good idea to merge ALL the typed parsing related logic into a single function because -

1) It's really hard to merge them all into one function and honestly it spoils readability of the code as there will be plenty of type checking both while parsing raw data or the parsing the buffer corresponding to the type. This way the code is much cleaner and easily readable as it is templated. 

2) This template will not be instantiated a lot of times -atmost a few times for the types that we need it for. So will not bloat up size too much. (Code readability vs size trade-off)

3) I would like to keep it consistent with tensor_proto_util.cc which has similar cleaner templated implementation